File: src/tests/control_stick_viz.test.ts
import { expect, test } from '@jest/globals';
import { Coord, controlStickViz, SlippiGame, getCoordListFromGame, toArrayBuffer, CheckResult } from '../index';
import * as fs from 'fs';
import * as path from 'path';

test('Test control stick viz (Sanity check)', async () => {
    const slpDir = path.join(__dirname, '../../test_data/nonlegal/analog/goomwave_uptilt_p1.slp')
    let data = fs.readFileSync(slpDir, null);
    let game = new SlippiGame(toArrayBuffer(data))
    expect(game).not.toBeNull()
    let gameCoordsPortOne: Coord[] = getCoordListFromGame(game, 0, true)

    let checkResult: CheckResult = controlStickViz(game, 0, gameCoordsPortOne)
    expect(checkResult.result).toEqual(false)
    expect(checkResult.violations[0].evidence.length).toEqual(4845)
})

File: libenforcer-wasm/tests/control_stick_viz_test.rs
//! Integration tests for control stick visualization check
//! Mirrors TypeScript tests in src/tests/control_stick_viz.test.ts
//! Test count: 1

#[path = "common/mod.rs"]
mod common;

#[cfg(not(target_arch = "wasm32"))]
use libenforcer_wasm::{checks::control_stick_viz, parser};
use peppi::io::slippi::de::read as read_slippi;
use std::io::Cursor;

use common::*;

#[test]
fn test_control_stick_viz_sanity_check() {
    let data = read_slp_file("nonlegal/analog/goomwave_uptilt_p1.slp");
    let game = read_slippi(&mut Cursor::new(&data), None).unwrap();
    let player_data = parser::extract_player_data(&game, 0).unwrap();

    let result = control_stick_viz::check(&player_data.main_coords);
    assert_eq!(result.result, false);
    assert_eq!(result.details.len(), 1);
    assert_eq!(
        result.details[0].evidence.len(),
        4845,
        "Expected 4845 evidence points for goomwave_uptilt_p1.slp"
    );
}
